### ISSUE TITLE
ttnn.softmax: "Static CBs exceed L1 size" exposed by VLLM Sampler

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#3028]
Blocks #3026 

**Issue:**

When removing `or True` in condition below so SamplingParams are no longer ignored, this error is seen from `ttnn.softmax` with CBs exceeding L1 space assertion firing.  It's a bit later in execution compared to when running with runtime debug enabled (#3027)

Needs debug and fixing.

```
    # @torch.compile(backend="tt", fullgraph=True, dynamic=False)
    def sample_from_logits(
        self, logits: torch.Tensor, sampling_metadata: TPUSupportedSamplingMetadata
    ) -> torch.Tensor:
        """
        Sample with xla-friendly function. This function is to be traced
        separately from `forward` for lighter compilation overhead.
        """
        # @LPanosTT: sampler functionalitty has issues currently, so we will always use greedy sampling
        if sampling_metadata.all_greedy or True:
            out_tokens = torch.argmax(logits, dim=-1, keepdim=True)
        else:
            out_tokens = self.sampler(logits, sampling_metadata).sampled_token_ids
        return out_tokens
```

**Repro:**

I was using f7d77e25 from Jan 26, though any recent commit would work.  Here I modified `runtime/lib/ttnn/program_executor.cpp` to change `Executing operation` print to be `LOG_INFO` for extra debug info.

[test_opt_generation_rm_hack.log](https://github.com/user-attachments/files/24927715/test_opt_generation_rm_hack.log)

Remove the `or True` argument above and run:

```
% cmake -G Ninja -B build |& tee cmake_cfg.log
% cmake --build build |& tee build.log
% TTXLA_LOGGER_LEVEL=DEBUG TTMLIR_RUNTIME_LOGGER_LEVEL=Debug pytest -svv tests/integrations/vllm_plugin/generative/test_opt_generation.py::test_opt_generation |& tee test_opt_generation_rm_hack.log

<snip>
            RuntimeTTNN |     INFO | Executing operation: %57 = "ttnn.softmax"(%56) <{compute_config = #ttnn.device_compute_kernel_config<math_fidelity = hifi4, fp32_dest_acc_en = true>, dimension = 1 : si32, numericStable = true}> : (tensor<1x50272xf32, #ttnn.ttnn_layout<(d0, d1) -> (d0, d1), <1x1>, memref<1x1571x!ttcore.tile<32x32, f32>, #ttnn.buffer_type<dram>>, <interleaved>>>) -> tensor<1x50272xf32, #ttnn.ttnn_layout<(d0, d1) -> (d0, d1), <1x1>, memref<1x1571x!ttcore.tile<32x32, f32>, #ttnn.buffer_type<dram>>, <interleaved>>> loc("divide.177")
    %c = stablehlo.constant dense<0> : tensor<i2026-01-29 03:28:46.025 | critical |          Always | TT_THROW: Statically allocated circular buffers on core range [(x=0,y=0) - (x=7,y=7)] grow to 19440768 B which is beyond max L1 size of 1499136 B (assert.hpp:104)

% grep Executing test_opt_generation_rm_hack.log | wc -l
8498
```

**Comment by kmabeeTT** [comment]
Marking this closed for now, the issue went away after rebasing... bisected to this commit in tt-xla (tt-mlir uplift):

```
2026-01-27 (7892d2ad) by GitHub (Author [ajakovljevic@tenstorrent.com](mailto:ajakovljevic@tenstorrent.com)) : Uplift third_party/tt-mlir to aef48b01b8dd14ddfe046a88760fc2a84093e3a5 2026-01-27 (#2955)
```

Bisected further to a tt-metal uplift:

```
2026-01-26 (aef48b01b) by GitHub (Author 157983820+vmilosevic@users.noreply.github.com) : Uplift third_party/tt-metal to a7d0da940efa838c201eb2b609ca2b4aab047832 2026-01-24 (#6742)
```

Didn't bisect further than that.
